### ISSUE TITLE
Librispeech RNN-T training does not look good with Espnet2

### ISSUE THREAD

**Issue by BSen007** [espnet/espnet#4536]
**Describe the bug**

**Basic environments:**
 - OS information: e.g., Linux 4.9.0-11-amd64 #1 SMP Debian 4.9.189-3+deb9u2 (2019-11-11) x86_64
 - python version: `3.7.5 (default, Oct 25 2019, 15:51:11)  [GCC 7.3.0]`
 - espnet version: espnet 202205
 - Git hash: 41ed22a0ad2ef57bf7c57ba1209dbef4e9f924cb
 - Commit date: Wed Jul 6 20:59:31 2022 +0900
 - pytorch version : pytorch 1.11.0
 

**Environments from `torch.utils.collect_env`:**
e.g.,
```
PyTorch version: 1.11.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.10

Python version: 3.7.5 (default, Oct 25 2019, 15:51:11)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-5.4.17-2036.100.6.1.el7uek.x86_64-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: 11.0.221
GPU models and configuration: 
GPU 0: A100-SXM4-40GB
GPU 1: A100-SXM4-40GB
GPU 2: A100-SXM4-40GB
GPU 3: A100-SXM4-40GB
GPU 4: A100-SXM4-40GB
GPU 5: A100-SXM4-40GB
GPU 6: A100-SXM4-40GB
GPU 7: A100-SXM4-40GB

Nvidia driver version: 450.51.06
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.5
[pip3] pytorch-lightning==1.6.4
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.1
[pip3] torch==1.11.0
[pip3] torch-complex==0.4.3
[pip3] torch-optimizer==0.3.0
[pip3] torchaudio==0.11.0
[pip3] torchmetrics==0.9.2
[pip3] warprnnt-pytorch==0.1
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py37h7f8727e_0  
[conda] mkl_fft                   1.3.1            py37hd3c417c_0  
[conda] mkl_random                1.2.2            py37h51133e4_0  
[conda] numpy                     1.21.4                   pypi_0    pypi
[conda] numpy-base                1.21.5           py37ha15fc14_3  
[conda] pytorch                   1.11.0          py3.7_cuda11.3_cudnn8.2.0_0    pytorch
[conda] pytorch-lightning         1.6.4                    pypi_0    pypi
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] pytorch-ranger            0.1.1                    pypi_0    pypi
[conda] pytorch-wpe               0.0.1                    pypi_0    pypi
[conda] torch-complex             0.4.3                    pypi_0    pypi
[conda] torch-optimizer           0.3.0                    pypi_0    pypi
[conda] torchaudio                0.11.0               py37_cu113    pytorch
[conda] torchmetrics              0.9.2                    pypi_0    pypi
[conda] warprnnt-pytorch          0.1                       dev_0    <develop>
```

**Task information:**
 - Task: ASR
 - Recipe: librispeech
 - ESPnet2

**To Reproduce**
I was trying out RNN-T recipe with librispeech (using libri data as well) and using same config file provided in the recipe. However, the training curve does not look good, especially the transducer loss. When training completed, I tried to decode and it takes a lot of time and getting only _AND_ . I wonder if I am making some mistake but chance is less as I am using the same recipe with same data. On the other hand, there is no issue while training conformer-CTC. 

![loss_ctc](https://user-images.githubusercontent.com/93305829/180923889-2deb541a-662c-48d1-be7b-674e1a33bdce.png)
![loss_transducer](https://user-images.githubusercontent.com/93305829/180923895-8e3e8d5f-36d9-46fb-99d2-42cadaba710e.png)

Suggestion please. Shall I try out some older commit with old pytorch. Is there any stability issue with RNN-T? 

1. move to a recipe directory, e.g., `cd egs/librispeech/asr1`
2. Use https://github.com/espnet/espnet/blob/master/egs2/librispeech/asr1/conf/tuning/transducer/train_conformer-rnn_transducer.yaml as config file
3. execute `run.sh` with specific arguments, e.g., `run.sh --stage 3 --ngpu 8`

**Logs**
Paste the error logs. If applicable, add screenshots to help explain your problem.

**Comment by b-flo** [comment]
Hi,

> Suggestion please. Shall I try out some older commit with old pytorch.

Hum, torch 1.11 is supported by our fork of `warp-transducer` so it shouldn't be an issue. It may be worth trying though because I'm suspecting an issue with `warp-transducer` given the loss curves is totally off.

> Is there any stability issue with RNN-T?

Not that I'm aware of, multiple groups (myself included) are currently using ESPnet2 Transducer without any issue.
However, the current training procedure is sensitive to some parameters (lr, accum grad, warmup). It can be difficult to tune in some instance depending on the number of GPUs and available VRAM compared to the original recipe.

**Comment by BSen007** [comment]
@b-flo thanks! can you please let me know which version of torch and espnet are you using? I can try the same!

**Comment by b-flo** [comment]
Sure! I'm using torch 1.8.1 with CUDA 11.1 for my current experiments

> and espnet

I can give a commit hash because I'm working on another branch right now but it's up to date with master.

**Comment by BSen007** [comment]
ok will plan to work with 1.8.1 to see the behavior! thanks :)

**Comment by BSen007** [comment]
No luck with 1.8.1 as well :(

**Comment by b-flo** [comment]
Honestly, I'm not sure what's the issue here.
For me, something went wrong with `warp-transducer` install as the loss doesn't make sense even if the model and training settings were way off. However, everything in your report seems OK except that there are different CUDA versions for building -11.3- and runtime -11.0-?
Could you share the logs for `warp-transducer` install, please?
